sdk/state/update_test.go

```diff
@@ -38,8 +38,12 @@ func TestLastConfirmedPayment(t *testing.T) {
 		RemoteEscrowAccount: localEscrowAccount,
 	})
 
+	// latest close agreement should be set during open steps
+	sendingChannel.latestCloseAgreement.Balance = Amount{Asset: NativeAsset{}}
+	receiverChannel.latestCloseAgreement.Balance = Amount{Asset: NativeAsset{}}
```

Comment:
out of curiosity -- why does the balance have on limit? Does that mean if someone pays the payment channel account with a native asset then that by default will be claimable by the initiator or remote based on who ends up with that escrow account?
and that this issue does not apply to non-native assets because of the trust limit that we set on the trustline?

Reply:
correct the `assetLimit` is just needed for the adding the trustlines to the escrow accounts. But that's a good point that we're not checking if a payment surpasses the escrow account `assetLimit`. Which could cause the closing transaction to fail if surpassing that assetLimit 🤔 

I'll add a ticket for that, because it needs some additional thinking through.

It may tie in with [this ticket](https://github.com/stellar/experimental-payment-channels/issues/92) for storing the open object on channel. So we can use it to check the channel assetLimit. Or maybe we should store on `Balance` as you said

Reply:
https://github.com/stellar/experimental-payment-channels/issues/93